x/launch/keeper/msg_server_settle_request_test.go

```diff
@@ -0,0 +1,268 @@
+				Approve:     true,
+			},
+			err: spnerrors.Critical(
+				fmt.Sprintf("account %s for chain %s not found", genesis, chainID)),
+		}, {
+			name: "test VestedAccount content",
+			msg: types.MsgSettleRequest{
+				ChainID:     chainID,
+				RequestID:   requests[2].RequestID,
+				Coordinator: sample.AccAddress(),
+				Approve:     true,
+			},
+		}, {
+			name: "test duplicated VestedAccount content",
+			msg: types.MsgSettleRequest{
+				ChainID:     chainID,
+				RequestID:   requests[2].RequestID,
+				Coordinator: sample.AccAddress(),
+				Approve:     true,
+			},
+			err: spnerrors.Critical(
+				fmt.Sprintf("account %s for chain %s already exist", vested, chainID)),
+		}, {
+			name: "test vested AccountRemoval content",
+			msg: types.MsgSettleRequest{
+				ChainID:     chainID,
+				RequestID:   requests[3].RequestID,
+				Coordinator: sample.AccAddress(),
+				Approve:     true,
+			},
+		}, {
+			name: "test not found vested AccountRemoval content",
+			msg: types.MsgSettleRequest{
+				ChainID:     chainID,
+				RequestID:   requests[3].RequestID,
+				Coordinator: sample.AccAddress(),
+				Approve:     true,
+			},
+			err: spnerrors.Critical(
+				fmt.Sprintf("account %s for chain %s not found", vested, chainID)),
+		}, {
+			name: "test GenesisValidator content",
+			msg: types.MsgSettleRequest{
+				ChainID:     chainID,
+				RequestID:   requests[4].RequestID,
+				Coordinator: sample.AccAddress(),
+				Approve:     true,
+			},
+		}, {
+			name: "test duplicated GenesisValidator content",
+			msg: types.MsgSettleRequest{
+				ChainID:     chainID,
+				RequestID:   requests[4].RequestID,
+				Coordinator: sample.AccAddress(),
+				Approve:     true,
+			},
+			err: spnerrors.Critical(
+				fmt.Sprintf("genesis validator %s for chain %s already exist", validator, chainID)),
+		}, {
+			name: "test ValidatorRemoval content",
+			msg: types.MsgSettleRequest{
+				ChainID:     chainID,
+				RequestID:   requests[5].RequestID,
+				Coordinator: sample.AccAddress(),
+				Approve:     true,
+			},
+		}, {
+			name: "test not found ValidatorRemoval content",
+			msg: types.MsgSettleRequest{
+				ChainID:     chainID,
+				RequestID:   requests[5].RequestID,
+				Coordinator: sample.AccAddress(),
+				Approve:     true,
+			},
+			err: spnerrors.Critical(
+				fmt.Sprintf("genesis validator %s for chain %s not found", validator, chainID)),
+		}, {
+			name: "invalid request",
+			msg: types.MsgSettleRequest{
+				ChainID:     chainID,
+				RequestID:   invalidContentID,
+				Coordinator: sample.AccAddress(),
+				Approve:     true,
+			},
+			err: spnerrors.Critical(
+				"no concrete type registered for type URL /tendermint.spn.launch.Request against interface *types.RequestContent"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			request, found := k.GetRequest(sdkCtx, tt.msg.ChainID, tt.msg.RequestID)
+			require.True(t, found)
+
+			err := applyRequest(sdkCtx, *k, &tt.msg, request)
```

Comment:
We should assert the request is correctly applied into the store when it is successful (account exists if add account, account is removed if no account, etc.)